pkgs/applications/audio/vcvrack/default.nix

```diff
@@ -0,0 +1,162 @@
+{ stdenv, fetchFromGitHub, fetchgit, makeWrapper, gcc, fetchpatch,
+  curl, glew, glfw, gnome2, jansson, libsamplerate, libzip,
+  pkgconfig, rtaudio, rtmidi,
+  ... }:
+
+with stdenv.lib;
+
+let
+  plugins = {
```

Comment:
I put the plugins into a separate `plugin.nix` file. I turned this file then into a function from plugins to derivation:
`vcvrack/default.nix`:
```
plugins:
stdenv.mkDerivation rec {
  name = "Rack-${version}";
  ...
}
```
`all-packages.nix`:
```
vcvrackWithPlugins = callPackage ../applications/audio/vcvrack { };
vcvrack = pkgs.vcvrackWithPlugins (import ../applications/audio/vcvrack/plugins.nix { inherit fetchFromGitHub; });
```

This allows people to add their own plugins by calling `vcvrackWithPlugins`.

Reply:
Good idea, I would like to be able to pick plugins like for vim or something.
Bad thing is that some of the plugins need the Rack repo as well.
I don't think a vcvrack without modules makes any sense. I think the vcvrack (withoutPlugins) should at least deliver the default repos.
I would like to deliver all the other modules as separate plugins. Something like
```
nixos.vcvrack
nixos.vcvrackPlugins.AS
nixos.vcvrackPlugins.Vult
...
````
what do you think ?

Reply:
> I don't think a vcvrack without modules makes any sense.
Yes this makes sense.

> I would like to deliver all the other modules as separate plugins.
How would vcvrack find the installed plugins? So far I have not found an easy way to compile plugins separate from vcvrack.

Reply:
> How would vcvrack find the installed plugins? So far I have not found an easy way to compile plugins separate from vcvrack.

I did not accomplish that task (of course).
But I thought of links. (of course we can't change the vcvrack, but maybe we use `/etc/` or something) 

something like `${pkgs.vcvrack}/usr/share/vcvrack/plugins -> /etc/vcvrack/plugins` and
than every plugin (e.g. AS plugin) creates `/etc/vcvrack/AS -> ${pkgs.vcvrackPlugins.AS}/usr/share/vcvrack/plugins/AS`

Reply:
> Yes this makes sense.

What do you want to do with a vcvrack without any plugins ? You can only look at the background image.

Reply:
@svenkeidel Would it be Ok, if we merge this pull-request with as few further changes, because @yegortimoshenko would have to spend time on reviewing it. And when it's merged you create a pull-request ? We could also talk on irc (i'm palo in the #nixos channel)

Reply:
>> Yes this makes sense.

> What do you want to do with a vcvrack without any plugins ? You can only look at the 
background image.

This was a misunderstanding :smile:, I was agreeing with you that vcvrack without any plugins does not make any sense.

Reply:
About the plugin problem. I don't know if _vcvrack finding plugins over the global filesystem_ is the most Nix way of doing things. I think that for packages that can be extended with plugins it is acceptable to take these as an input argument, for example, [emacsWithPackages](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/emacs/wrapper.nix). This gives the users more flexibility to instantiate this package multiple times with different sets of plugins. This flexibility probably will not be needed in case of vcvrack, however, I think it is good practice.

Based this plea, I would like to refine my proposal from above to support a standard set of plugins:

vcvrack/default.nix:
```
{ stdenv, ... }:

plugins:
stdenv.mkDerivation rec {
  name = "Rack-${version}";
  ...
}
```

all-packages.nix:
```
vcvrackWithPlugins = callPackage ../applications/audio/vcvrack { };
vcvrack = userPlugins:
  let corePlugins =  import ../applications/audio/vcvrack/plugins-core.nix { inherit fetchFromGitHub; }
  in; pkgs.vcvrackWithPlugins (vcvrackCorePlugins // userPlugins);
```

Reply:
@svenkeidel : Yeah that sounds also good. But I still would like to "just" get this pr done (also because it is so old) and solve your issues in another pull-request right after this thing got merged.